test/performance/compare-benchmarks.ts

```diff
@@ -0,0 +1,93 @@
+import fs = require('fs')
+import asTable = require('as-table')
+
+interface ResultSuite {
+  suiteName: string,
+  results: { testName: string, time: number }[],
+}
+
+(() => {
+  try {
+    const args = process.argv.slice(2)
+
+    if (!args || args.length !== 3) {
+      console.log('Usage:\n$ npm run benchmark:compare-benchmarks base-benchmarks.json current-change-benchmarks.json output-file.md')
+      return
+    }
+
+    const inputFilenames = args.slice(0, 2)
+    const outputFilename = args[2]
+
+    const resultSuites: ResultSuite[] = inputFilenames.map(readResultSuiteFromFile)
+    const benchmarkTable = buildBenchmarkTable(resultSuites)
+    writeTableToFile(benchmarkTable, outputFilename)
+  } catch (err) {
+    console.error(err)
+  }
+})()
+
+function writeTableToFile(tableData: { [key: string]: string | number }[], filename: string): void {
+  const tableRenderer = asTable.configure({ delimiter: ' | ', right: true })
+  const renderedTable = `# Performance comparison\n\n\`\`\`\n${tableRenderer(tableData)}\n\`\`\`\n`
```

Comment:
Since this job will not run on every commit, could the results report contain the Git SHA of the versions used for the comparison?

For reference, the Codecov comment contains information about the version (see the below screenshot).

I don't think we need the PR number or the branch name there. Just a simple Git SHA will suffice for me.

<img width="657" alt="Screen Shot 2022-08-23 at 3 05 32 PM" src="https://user-images.githubusercontent.com/566463/186165468-d2c47ad1-db1c-4545-973f-81aa3b1e9b94.png">